### PR TITLE
[Merged by Bors] - feat : port Algebra.Star.Pointwise

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
@Ruben-VandeVelde Thanks for the fix!

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Data/Set/Pointwise/Basic.lean [hunk 1]
```diff
@@ -297,7 +297,7 @@ protected def mul : Mul (Set α) :=
 scoped[Pointwise] attribute [instance] Set.mul Set.add
 
 @[to_additive (attr := simp)]
-theorem image2_mul : image2 Mul.mul s t = s * t :=
+theorem image2_mul : image2 (· * ·) s t = s * t :=
```
Comment (jcommelin) [inline Mathlib/Data/Set/Pointwise/Basic.lean:300]: Why is this change needed?

**Review by jcommelin — COMMENTED** [review]
1 question, otherwise LGTM

File: Mathlib/Data/Set/Pointwise/Basic.lean (hunk 1 above)
Comment (xroblot) [reply Mathlib/Data/Set/Pointwise/Basic.lean:300]: Otherwise, the proof of `star_mul` gets more complicated since at some point, one needs to prove that 
```(fun a b : α ↦ (Mul.mul a b)⋆) = (fun a b : α ↦ b⋆ * a⋆) ```
Note that I was the one that did the porting of `Data/Pointwise/Basic` so it is likely that I wrote the wrong statement. The original mathlib3 lemma is:
```lemma image2_mul : image2 has_mul.mul s t = s * t := rfl```

This change was suggested by @Ruben-VandeVelde

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge
